src/utils_request.js

```diff
@@ -300,11 +331,17 @@ function normalizeSslErrorHandling(ignoreSslErrors, gotScrapingOptions) {
  * @private
  */
 function ensureCorrectHttp2Headers(gotScrapingOptions) {
-    if (gotScrapingOptions.http2) {
-        delete gotScrapingOptions.headers?.connection;
-        delete gotScrapingOptions.headers?.Connection;
-        delete gotScrapingOptions.headers?.host;
-        delete gotScrapingOptions.headers?.Host;
+    if (gotScrapingOptions.http2 && gotScrapingOptions.headers) {
+        gotScrapingOptions.headers = { ...gotScrapingOptions.headers };
+
+        // eslint-disable-next-line no-restricted-syntax, guard-for-in
+        for (const key in gotScrapingOptions.headers) {
+            const lkey = key.toLowerCase();
+
+            if (lkey === 'connection' || lkey === 'host') {
+                delete gotScrapingOptions.headers[key];
+            }
+        }
```

Comment:
I don't think this is more readable than the previous version, but yeah, it does capture stuff like `coNNection` 👍